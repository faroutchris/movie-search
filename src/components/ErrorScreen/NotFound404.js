import React from "react";
import { Link } from "react-navi";
const NotFound = () => (
  <div>
    <h1>404 Not found</h1>
    <Link href="/">Back to start page</Link>
  </div>
);

export default NotFound;
