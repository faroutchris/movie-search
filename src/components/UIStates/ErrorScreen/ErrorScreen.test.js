import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import ErrorScreen from "./ErrorScreen";

describe("ErrorScreen", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<ErrorScreen>No error</ErrorScreen>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("ErrorScreen shows a fallback UI when a component has crashed", () => {
    const ComponentWillCrash = () => {
      throw new Error("I crashed");
    };

    const component = TestRenderer.create(
      <ErrorScreen inTestMode={true}>
        <ComponentWillCrash />
      </ErrorScreen>
    );

    const tree = component.toJSON();
    console.info(
      "\n\n========================================\n\n" +
        "THE ERROR ABOVE IS EXPECTED IN THE TESTS" +
        "\n\n========================================\n\n"
    );
    expect(tree).toMatchSnapshot();
  });
});
