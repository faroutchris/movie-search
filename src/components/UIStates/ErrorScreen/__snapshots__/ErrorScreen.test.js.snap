// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen ErrorScreen shows a fallback UI when a component has crashed 1`] = `
<h1>
  Something went wrong.
</h1>
`;
